plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryMetadata.java

```diff
@@ -393,16 +408,29 @@ private void createTable(ConnectorSession session, ConnectorTableMetadata tableM
             throw new SchemaNotFoundException(schemaName);
         }
 
-        List<Field> fields = tableMetadata.getColumns().stream()
-                .map(column -> toField(column.getName(), column.getType(), column.getComment()))
-                .collect(toImmutableList());
+        ImmutableList.Builder<Field> fields = ImmutableList.builder();
+        ImmutableList.Builder<String> columnsNames = ImmutableList.builder();
+        ImmutableList.Builder<Type> columnsTypes = ImmutableList.builder();
```

Comment:
nit: since we know the size we can use `builderWithExpectedSize` (or whatever it's called).